### PR TITLE
Refactoring mia's

### PR DESCRIPTION
**Pull request by cedricdcc**
Redone the Core approach on how the widget is constructed.

### CODE REVIEW

File: src/affordance_entity.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import { createEmptyStore, storeSize , getLinkedDataNQuads, addDataToStore} from "./linked_data_store.js";
+import { getBoundryInfo, getInfoPopup } from "./info_extraction.js";
+import Popup from "./popup.js";
+
+export default class AffordanceEntity {
+    constructor(affordance) {
+        this.uri = affordance.href;
+        this.id = 'id-' + Math.random().toString(36).slice(2, 11); // todo: check if this is really needed in the end
+        this.node = affordance;
+        this.store = createEmptyStore();
```
Comment (marc-portier) [inline src/affordance_entity.js:12]: what is this store about? every variable is there to 'store' something -- better to consider a name that hints at what it is about.

File: src/affordance_entity.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        this.checked = false;
+        this.icon = null;
+
+        //add the id to the node as the mia_entity_id attribute
+        this.node.setAttribute('mia_entity_id', this.id);
+
+        //do check if uri is from marine regions or marine info or any other known source with linked data
+        this.checkUri();
+        //perform span manipulation based on info from the initial class creation
+        this.NodeManipulations();
```
Comment (marc-portier) [inline src/affordance_entity.js:22]: odd uppercase method name `NodeManipulations` -- suggests static which would be awkward

also `manipulations` is generic `doing` `function` `process` `method` all are not adding clear semantics about the actual purpose of this method -- so unhelpfully abstract

File: src/affordance_entity.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        //on hover do stuff
+        this.node.addEventListener('mouseover', (event) => {
+            logger.log(this);
+            //check if the entity needs to get linked data by checking the store size
+            if (storeSize(this.store) === 0 && !this.checked) {
```
Comment (marc-portier) [inline src/affordance_entity.js:27]: odd external method? OO would suggest `this.store.size()` ??

**Review by marc-portier — CHANGES_REQUESTED** [review]
sorry, limited time, so only ran over it quickly / diagonally
already some stuff to consider though

+ recommend finding an extra buddy in our team for this

**Review by marc-portier — APPROVED** [review]
out of interest -- do we have some automated tests for this module - if not now, somewhere in the planning?

**Comment by cedricdcc** [comment]
> out of interest -- do we have some automated tests for this module - if not now, somewhere in the planning?

These will be added in the future

**Pull request merged**